src/main/java/reposense/parser/CsvParser.java

```diff
@@ -211,22 +223,51 @@ private String getRowContentAsRawString(final CSVRecord record) {
     }
 
     /**
-     * Checks if {@code possibleHeader} contains the expected number of columns.
-     * @throws InvalidCsvException if {@code possibleHeader} does not have as many columns as expected.
+     * Generates map of column header to position number for input {@code possibleHeader}.
+     * @throws InvalidCsvException if {@code possibleHeader} does not contain all the mandatory headers.
      */
     private void validateHeader(String[] possibleHeader) throws InvalidCsvException {
-        int actualNumberOfColumns = possibleHeader.length;
-        if (actualNumberOfColumns != expectedHeaderSize) {
-            throw new InvalidCsvException(String.format(
-                    MESSAGE_WRONG_HEADER_SIZE, csvFilePath.getFileName(), actualNumberOfColumns,
-                    expectedHeaderSize));
+        headerSize = possibleHeader.length;
+        for (int i = 0; i < headerSize; i++) {
+            String possible = possibleHeader[i].trim();
+            for (String parsedHeader : mandatoryAndOptionalHeaders()) {
+                if (possible.equalsIgnoreCase(parsedHeader)) {
+                    if (headerMap.containsKey(parsedHeader)) {
+                        logger.warning(String.format(
+                                MESSAGE_DUPLICATE_COLUMN_HEADER, parsedHeader, csvFilePath.getFileName()));
```

Comment:
We should add a test where the CSV file has multiple headers of the same name.

Reply:
I have added 2 tests, for case-sensitive and case-insensitive duplicates.